src/id/mangatale/src/eu/kanade/tachiyomi/extension/id/mangatale/Ikiru.kt

```diff
@@ -1,42 +1,241 @@
+import keiyoushi.utils.tryParse
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MultipartBody
 import okhttp3.OkHttpClient
-import okhttp3.ResponseBody.Companion.toResponseBody
+import okhttp3.Request
+import okhttp3.RequestBody
 import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
 
-class Ikiru : MangaThemesia("Ikiru", "https://id.ikiru.wtf", "id") {
-
+class Ikiru() : ParsedHttpSource(), ConfigurableSource {
+    // Formerly "MangaTale"
     override val id = 1532456597012176985
 
+    override val name = "Ikiru"
+    private val defaultBaseUrl: String = "https://01.ikiru.wtf"
+    override val lang = "id"
+    override val supportsLatest = true
+
+    override val baseUrl: String get() = when {
+        isCi -> defaultBaseUrl
+        else -> preferences.prefBaseUrl
+    }
+
+    private val isCi = System.getenv("CI") == "true"
+
     override val client: OkHttpClient = super.client.newBuilder()
         .rateLimit(12, 3)
-        .addInterceptor { chain ->
-            val response = chain.proceed(chain.request())
-            val mime = response.headers["Content-Type"]
-            if (response.isSuccessful) {
-                if (mime != "application/octet-stream") {
-                    return@addInterceptor response
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    // Popular
+    override fun popularMangaRequest(page: Int): Request = searchMangaRequest(page, "", FilterList())
+
+    override fun popularMangaSelector() = searchMangaSelector()
+
+    override fun popularMangaFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun popularMangaNextPageSelector() = searchMangaNextPageSelector()
+
+    // Latest
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/latest-update/?the_page=$page", headers)
+    }
+
+    override fun latestUpdatesSelector() = "#search-results > div:not(.col-span-full)"
+
+    override fun latestUpdatesFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = "#search-results ~ div.col-span-full a:has(svg):last-of-type"
+
+    // Search
+    private var searchNonce: String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        // TODO: Filter
+
+        if (searchNonce.isNullOrEmpty()) {
+            try {
+                val response = client.newCall(
+                    GET("$baseUrl/ajax-call?type=search_form&action=get_nonce", headers),
+                ).execute()
+                val document = response.asJsoup()
+                searchNonce = document.select("input[name=search_nonce]").attr("value")
+            } catch (e: Exception) {
+                Log.e(name, "Error fetching nonce: ", e)
+            }
+        }
```

Comment:
```suggestion
        if (searchNonce.isNullOrEmpty()) {
            val document = client.newCall(
                GET("$baseUrl/ajax-call?type=search_form&action=get_nonce", headers),
            ).execute().asJsoup()
            searchNonce = document.selectFirst("input[name=search_nonce]")!!.attr("value")
        }
```
the extension should fail when nonce isn't found, since the api call will not work without it. so don't wrap in try catch. also fail if it isn't found